src/WebJobs.Extensions.Http/Extensions/HttpRequestExtensions.cs

```diff
@@ -56,32 +57,59 @@ public static bool IsJsonContentType(this HttpRequest request)
                 string.Equals(headerValue.MediaType, "application/json", StringComparison.OrdinalIgnoreCase);
         }
 
-        public static ClaimsIdentity GetAppServiceIdentity(this HttpRequest request)
+        public static bool TryGetAuthIdentity(this HttpRequest request, AuthIdentityEnum authIdentity, out ClaimsIdentity claimsIdentity)
```

Comment:
This is a breaking change to a public API - is this a necessary change? Who is calling this new API -  I don't see it in your host PR.

Reply:
Indeed. We can keep the existing API and call into this with the original behavior.